UTAPI-116: Check stderr and fix deprecation warnings

BourgoisMickael opened the pull request:
It floods the stderr logs of cloudserver and utapi

```
(node:4634) [DEP0174] DeprecationWarning: Calling promisify on a function that returns a Promise is likely a mistake.
```

BourgoisMickael committed:
UTAPI-116: Suppress aws sdk v2 warnings

BourgoisMickael committed:
UTAPI-116: Collect stderr and check for logs

bert-e commented:
# Hello bourgoismickael,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/utapi/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

BourgoisMickael committed:
UTAPI-116: Fix node22 warning DEP0174 promisify

It floods the stderr logs of cloudserver and utapi

```
(node:4634) [DEP0174] DeprecationWarning: Calling promisify on a function that returns a Promise is likely a mistake.
```

BourgoisMickael committed:
UTAPI-116: Fix warning MaxListenersExceededWarning

In the cleanup of functional/v2/task/testCreateSnapshot.js

BourgoisMickael committed:
UTAPI-116: Ignore DEP0040 punycode from oas-tools

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -18,6 +18,11 @@ on:
+
+env:
+  AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE: '1'
+  # NODE_OPTIONS: '--trace-warnings'
```
```suggestion
```

BourgoisMickael replied:
I prefer to keep so we can easily uncomment to have stack trace if a warning appear. 

I commented because there is currently a warning ignored

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -137,6 +142,10 @@ jobs:
       run: ${{ matrix.test.command }}
       env: ${{ matrix.test.env }}
+    - name: Print stderr logs
+      run: grep -H "" *.stderr.log || true
```
Is this like for debugging purposes? If so let's make sure it is run everytime.
```suggestion
      run: grep -H "" *.stderr.log || true
      if: always()
```

BourgoisMickael replied:
It's for debugging the warnings (from the step below). 

If there is a test error before there is no need to print those stderr, they already appear in regular CI output

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -242,6 +251,12 @@ jobs:
         tmate-server-ed25519-fingerprint: ${{ secrets.TMATE_SERVER_ED25519_FINGERPRINT }}
       if: ${{ ( github.event.inputs.debug == true || github.event.inputs.debug == 'true' ) }}
+    - name: Print stderr logs
+      run: grep -H "" *.stderr.log || true
```
```suggestion
      run: grep -H "" *.stderr.log || true
      if: always()
```

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -358,3 +373,7 @@ jobs:
         tmate-server-ed25519-fingerprint: ${{ secrets.TMATE_SERVER_ED25519_FINGERPRINT }}
       if: ${{ ( github.event.inputs.debug == true || github.event.inputs.debug == 'true' ) }}
+    - name: Print stderr logs
+      run: grep -H "" *.stderr.log || true
```
```suggestion
      run: grep -H "" *.stderr.log || true
      if: always()
```

jonathan-gramain commented on .github/workflows/tests.yaml:
```diff
@@ -137,6 +142,10 @@ jobs:
       env: ${{ matrix.test.env }}
+    - name: Print stderr logs
+      run: grep -H "" *.stderr.log || true
+    - name: Error if stderr logs are not empty
```
Maybe a bit extreme to fail the test? IDK, just a personal feeling, it's useful to have feedback on any new stderr message, maybe we can add a `continue-on-error: true` so it's visible but doesn't outright fail the test if the overall test suite is successful.

BourgoisMickael replied:
That's the idea to catch immediately a warning, so we are not surprised later.

Then if we don't plan on fixing the warning, we can filter it out from the logs ([38badb7](https://github.com/scality/utapi/pull/1318/commits/38badb76a0402ff224b46f0b9b85bca94e9f3273))

A `continue-on-error: true` could be added at any moment if there are warnings that can't be fixed to pass the tests, but ideally I'd like to fail to make sure we notice it. 

And I'd like to have this in other components as well in long term

jonathan-gramain commented on libV2/redis.js:
```diff
@@ -15,6 +15,49 @@ const moduleLogger = new LoggerContext({
+ * @param {Function} originalFn The function to promisify.
+ * @returns {Function} A function that returns a promise.
+ */
+function flexiblePromisify(originalFn) {
```
Is this a standard, proven function?
- if so, can you provide the ref
- if not, can you add unit tests to make sure it works as designed?

BourgoisMickael replied:
Nothing standard, made with gemini

here are unit tests: [7a12ad2](https://github.com/scality/utapi/pull/1318/commits/7a12ad252e1f37653096f67088a8311bfe4b7d4e)

BourgoisMickael committed:
UTAPI-116: Unit tests for flexiblePromisify

BourgoisMickael commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.1`



Please check the status of the associated issue UTAPI-116.

Goodbye bourgoismickael.




*The following options are set:* **approve**
